src/consultations/consultations.controller.ts

```diff
@@ -449,4 +450,94 @@ export class ConsultationsController {
     )
   }
+
+  @JwtAuth()
+  @RequiredPermission(Permissions.JoinConsultation)
+  @Get(':id/ics')
+  async getCalendar(
+    @Param('id', ParseIntPipe) id: number,
+    @CurrentUser() user: UserEntity,
+    @Res() res: Response,
+  ): Promise<StreamableFile> {
+    const consultation = await this.consultationsService.findOne(id, user)
+
+    const startString =
+      consultation.startDate.getFullYear().toString() +
+      ('0' + consultation.startDate.getMonth().toString()).slice(-2) +
+      ('0' + consultation.startDate.getDate().toString()).slice(-2) +
+      `T` +
+      ('0' + consultation.startDate.getHours().toString()).slice(-2) +
+      ('0' + consultation.startDate.getMinutes().toString()).slice(-2) +
+      ('0' + consultation.startDate.getSeconds().toString()).slice(-2)
+    const endString =
+      consultation.endDate.getFullYear().toString() +
+      ('0' + consultation.endDate.getMonth().toString()).slice(-2) +
+      ('0' + consultation.endDate.getDate().toString()).slice(-2) +
+      `T` +
+      ('0' + consultation.endDate.getHours().toString()).slice(-2) +
+      ('0' + consultation.endDate.getMinutes().toString()).slice(-2) +
+      ('0' + consultation.endDate.getSeconds().toString()).slice(-2)
+    const eventText =
+      `BEGIN:VCALENDAR
+    PRODID:-//Google Inc//Google Calendar 70.9054//EN
+    VERSION:2.0
+    CALSCALE:GREGORIAN
+    METHOD:REQUEST
+    BEGIN:VTIMEZONE
+    TZID:Europe/Budapest
+    X-LIC-LOCATION:Europe/Budapest
+    BEGIN:DAYLIGHT
+    TZOFFSETFROM:+0100
+    TZOFFSETTO:+0200
+    TZNAME:CEST
+    RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU
+    END:DAYLIGHT
+    BEGIN:STANDARD
+    TZOFFSETFROM:+0200
+    TZOFFSETTO:+0100
+    TZNAME:CET
+    DTSTART:19701025T030000
+    RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU
+    END:STANDARD
+    END:VTIMEZONE
+    BEGIN:VEVENT
+    DTSTART;TZID=Europe/Budapest:` +
+      startString +
+      `\nDTEND;TZID=Europe/Budapest:` +
+      endString +
+      `\nX-MICROSOFT-CDO-OWNERAPPTID:-1911733949
+    DESCRIPTION: ` +
+      consultation.descMarkdown.toString() +
+      `\nLOCATION:` +
+      consultation.location.toString() +
+      `
+    SEQUENCE:0
+    STATUS:CONFIRMED
+    SUMMARY:` +
+      consultation.name.toString() +
+      `
+    TRANSP:OPAQUE
+    END:VEVENT
+    END:VCALENDAR
+    `
+
+    if (!consultation.archived) {
+      res.set({
+        'Content-Disposition': 'attachment; fileame"konzi_calendar.ics"',
+        'Content-type': 'text/plain',
+      })
+      const steamableFile = new StreamableFile(Readable.from(eventText))
+      steamableFile.setErrorHandler((err, response) => {
+        response.statusCode = HttpStatus.NOT_FOUND
+        response.send(
+          JSON.stringify({
+            statusCode: HttpStatus.NOT_FOUND,
+            message: 'Az esemény archiválva0 lett',
+          }),
+        )
+      })
+      return steamableFile
+    }
+    throw new HttpException('Az esemény archiválva lett', HttpStatus.NOT_FOUND)
+  }
```

Comment:
we don't have to worry about archiving, on the frontend users will only be able to call this endpoint if the konzi hasn't started yet, so it won't be archived. And since we're not actually storing the ics file on the server, it's not an issue if it can still be downloaded after the konzi using the REST API.